src/main/webapp/js/finscholar/commonlistview/commonlistview.js

```diff
@@ -163,11 +170,13 @@ class CommonListView extends BasicView {
       const dataBatch = await this.dataHandler_.getNextBatch(
           this.optionTag_, this.batch_, numberOfItems, this.idOfLastItem_);
       const dataList = dataBatch ? dataBatch[ITEM] : undefined;
-      this.idOfLastItem_ =
-          dataList ? dataList[dataList.length - 1][0] : EMPTY_STRING;
-      if (!dataBatch || !dataList) {
+      if (!dataBatch || !dataList || dataList.length == 0) {
+        this.statusBar_.innerHTML = endoflist();
+        this.loading_ = false;
```

Comment:
This property is not defined in the constructor.

Please add a jsdoc and define it in the constructor.